### ISSUE TITLE
Installing with METIS support without sudo

### ISSUE THREAD

**Issue by legitmaxwu** [rusty1s/pytorch_sparse#107]
I would like to install torch_sparse with METIS support. I am able to build metis--however, I am able to run `sudo make install` because I don't have privileges. Is there a way around this? (i.e provide an installation path other than the default `/usr` directory)

When I run `pip install torch_sparse`, I get the following as expected:
```
...x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lmetis
  collect2: error: ld returned 1 exit status
```

Thanks!

**Comment by rusty1s** [comment]
You can add the path containing the `metis.so` file to `$LD_LIBRARY_PATH`. This way, no `sudo` is required to place the library into `usr/lib`.

**Comment by legitmaxwu** [comment]
Thanks so much for the tip! It led me away from a deep dark path and I was able to successfully install torch_sparse.

Apparently `$LD_LIBRARY_PATH` is used during runtime but not during build time. In build time linking, the `-L` flag is used instead to add directories to search. So in the error message, wherever the `-L` flag was pointing to is where the libraries need to go.

In my case, I was using a conda environment, so what fixed things for me was: (I created an env called `gnn`)

1. Copying `libmetis.so` to `.../miniconda3/envs/gnn/lib`
2. Copying `metis.h` to `.../miniconda3/envs/gnn/include`

Not sure if there's a more elegant solution, but this worked for me!
